packages/grafana-ui/src/components/Table/styles.ts

```diff
@@ -1,140 +1,163 @@
-        align-items: center;
-        height: 100%;
+      }
 
-        &:last-child {
-          border-right: none;
+      &:hover {
+        overflow: visible;
+        width: auto !important;
+        box-shadow: 0 0 2px ${theme.colors.formFocusOutline};
+        background: ${background ?? rowHoverBg};
+        z-index: 1;
 
-          > div {
-            padding-right: ${scollbarWidth + padding}px;
-          }
+        .cell-filter-actions  {
+          display: inline-flex;
         }
-      `,
-      tableCellLink: css`
-        text-decoration: underline;
-      `,
-      tableCell: css`
-        padding: ${padding}px;
-        text-overflow: ellipsis;
-        white-space: nowrap;
-        overflow: hidden;
-        flex: 1;
-      `,
-      overflow: css`
-        overflow: hidden;
-        text-overflow: ellipsis;
-      `,
-      resizeHandle: css`
-        label: resizeHandle;
-        cursor: col-resize !important;
-        display: inline-block;
-        background: ${resizerColor};
-        opacity: 0;
-        transition: opacity 0.2s ease-in-out;
-        width: 8px;
-        height: 100%;
-        position: absolute;
-        right: -4px;
-        border-radius: 3px;
-        top: 0;
-        z-index: ${theme.zIndex.dropdown};
-        touch-action: none;
+      }
+    `;
+  };
 
-        &:hover {
-          opacity: 1;
-        }
+  return {
+    theme,
+    cellHeight,
+    buildCellContainerStyle,
+    cellPadding,
+    cellHeightInner: bodyFontSize * lineHeight,
+    rowHeight: cellHeight + 2,
+    table: css`
+      height: 100%;
+      width: 100%;
+      overflow: auto;
+      display: flex;
+    `,
+    thead: css`
+      label: thead;
+      height: ${cellHeight}px;
+      overflow-y: auto;
+      overflow-x: hidden;
+      background: ${headerBg};
+      position: relative;
+    `,
+    headerCell: css`
+      padding: ${cellPadding}px;
+      overflow: hidden;
+      white-space: nowrap;
+      color: ${colors.textBlue};
+      border-right: 1px solid ${theme.colors.panelBg};
+      display: flex;
+
+      &:last-child {
+        border-right: none;
+      }
+    `,
+    headerCellLabel: css`
+      cursor: pointer;
+      white-space: nowrap;
+      overflow: hidden;
+      text-overflow: ellipsis;
+      display: flex;
+      margin-right: ${theme.spacing.xs};
+    `,
+    cellContainer: buildCellContainerStyle(),
+    cellText: css`
+      cursor: text;
+      overflow: hidden;
+      text-overflow: ellipsis;
+      user-select: text;
+      white-space: nowrap;
+    `,
+    cellLink: css`
+      cursor: pointer;
+      overflow: hidden;
+      text-overflow: ellipsis;
+      user-select: text;
+      white-space: nowrap;
+      text-decoration: underline;
+    `,
+    headerFilter: css`
+      label: headerFilter;
+      cursor: pointer;
+    `,
+    row: css`
+      label: row;
+      border-bottom: 1px solid ${borderColor};
+
+      &:hover {
+        background-color: ${rowHoverBg};
+      }
+    `,
```

Comment:
I would move this to table.tsx.

Reply:
there are a lot of variables that are shared in all the table styles, do not want to duplicate them in multiple places  (like rowHoverBg, is used in cell style)

Reply:
Makes sense! Missed that one. So then I would categorized those into the shared category and keep them in the `styles.ts`.

Reply:
Another approach that could be worth thinking about would be to move the shared colors etc. to the theme. Then it is also reusable for plugins if they want to extend the table cells in some way. But not sure it is a valid option.

Reply:
Yes, that could be a good idea. Most are high-level colors from the theme already, and trying to limit component variables in the theme. But if we had another use of these in other components or plugins then that would be good idea